api/src/routes/auth.ts

```diff
@@ -65,39 +69,62 @@ const defaultUser = {
   username: ''
 };
 
+const getEmailFromAuth0 = async (req: FastifyRequest) => {
+  const auth0Res = await fetch(`https://${AUTH0_DOMAIN}/userinfo`, {
+    headers: {
+      Authorization: req.headers.authorization ?? ''
+    }
+  });
+
+  if (!auth0Res.ok) {
+    req.log.error(auth0Res);
+    throw new Error('Invalid Auth0 Access Token');
+  }
+
+  const { email } = (await auth0Res.json()) as { email: string };
+  return email;
+};
+
+const findOrCreateUser = async (fastify: FastifyInstance, email: string) => {
+  const existingUser = await fastify.prisma.user.findFirst({
+    where: { email },
+    select: { id: true }
+  });
+  return (
+    existingUser ??
+    (await fastify.prisma.user.create({
+      data: { ...defaultUser, email },
+      select: { id: true }
+    }))
+  );
+};
```

Comment:
Yes, that's expected - maybe add some to-do comments? I left this as a normal comment (not RFC), in case you did not notice :), so I will later in case you don't want to.